src/Plugin/QuickForm/SaatenUnionSpraying.php

```diff
@@ -0,0 +1,746 @@
+      'required' => TRUE,
+      'type' => ['#value' => 'material'],
+      'measure' => ['#value' => 'rate'],
+      'units' => ['#options' => $product_units_options],
+    ]);
+
+    $form['total_product_quantity'] = $this->buildQuantityField([
+      'title' => $this->t('Total Product Quantity'),
+      'description' => $this->t('The total amount of product required to cover the field area(s).'),
+      'required' => TRUE,
+      'type' => ['#value' => 'material'],
+      'measure' => ['#value' => 'rate'],
+      'units' => ['#options' => $product_units_options],
+    ]);
+
+    // Water Volume remaining.
+    $water_volume_units_options = [
+      'l' => 'l',
+      'gal' => 'gal',
+    ];
+    $water_volume_remaining = [
+      'title' => $this->t('Water Volume'),
+      'description' => $this->t('The total amount of water required to cover the field area(s).'),
+      'measure' => ['#value' => 'volume'],
+      'units' => ['#options' => $water_volume_units_options],
+    ];
+    $form['water_volume']  = $this->buildQuantityField($water_volume_remaining);
+
+    // Area sprayed.
+    $area_sprayed_units_options = [
+      'm2' => 'm2',
+      'ha' => 'ha',
+    ];
+    $area_sprayed = [
+      'title' => $this->t('Area sprayed'),
+      'description' => $this->t('The total area being sprayed.'),
+      'measure' => ['#value' => 'area'],
+      'units' => ['#options' => $area_sprayed_units_options],
+    ];
+    $form['area'] = $this->buildQuantityField($area_sprayed);
+
+    // Wind speed.
+    $wind_speed_units_options = [
+      'kph' => 'kph',
+      'mph' => 'mph',
+    ];
+    $wind_speed = [
+      'title' => $this->t('Wind speed'),
+      'description' => $this->t('The maximum wind speed during spraying.'),
+      'measure' => ['#value' => 'ratio'],
+      'units' => ['#options' => $wind_speed_units_options],
+      'required' => TRUE,
+    ];
+    $form['wind_speed'] = $this->buildQuantityField($wind_speed);
+
+    // Wind direction.
+    $wind_directions = [
+      $this->t('North'),
+      $this->t('South'),
+      $this->t('East'),
+      $this->t('West'),
+      $this->t('North East'),
+      $this->t('North West'),
+      $this->t('South East'),
+      $this->t('South West'),
+    ];
+
+    $wind_direction_options = array_combine($wind_directions, $wind_directions);
+    $form['wind_direction'] = [
+      '#type' => 'select',
+      '#title' => $this->t('Wind Direction'),
+      '#description' => $this->t('The dominant wind direction during spraying. Please select the general direction the wind is coming from.'),
+      '#options' => $wind_direction_options,
+      '#required' => TRUE,
+    ];
+
+    $form['temperature'] = $this->buildQuantityField([
+      'title' => $this->t('Temperature (C)'),
+      'description' => $this->t('The average temperature during spraying.'),
+      'measure' => ['#value' => 'temperature'],
+      'units' => ['#options' => ['c' => 'C']],
```

Comment:
Whenever there is only a single `units` option, you can just specify `'units' => ['#value' => 'c']`. This will improve the UI and not show a dropdown when there is only a single option.

Reply:
@paul121 this has been fixed.